### ISSUE TITLE
Directory support for getFixtures

### ISSUE THREAD

**Issue by Soullivaneuh** [hautelook/AliceBundle#116]
Could be nice to have something like this:

```php
public function getFixtures()
{
    return [
        __DIR__.'/fixtures.yml',
        __DIR__.'/accounting.yml',
        __DIR__.'/user_browser.yml',
        __DIR__,
    ];
}
```

In this case, this will load those three fixtures files first and then, the rest present on this directories and it sub-directories.

What do you think?

**Comment by theofidry** [comment]
I would prefer to keep things as it is. If you want to return all the files of a folder you can already do it:

```php
public function getFixtures()
{
    $finder = new \Symfony\Component\Finder\Finder();
    $finder->in(__DIR__.'/products')->depth(0)->files()->name('*.yml')->sortByName();
    
    return iterator_to_array($finder);
}
```

I would prefer that since the fixtures loader may support different kind of files format (other than YAML and PHP) and have custom parsers to add support to other files.

If you and some other folk feel as if it would be far more convenient, we'll change that.

**Comment by Soullivaneuh** [comment]
Nice tip, but if I want to specify some files in priority before the directory itself?

**Comment by theofidry** [comment]
Actually the order of files is completely random and may change form a machine to another with the exact same configuration, so you should definitely not rely on such thing. But why would you need to set an order?

**Comment by theofidry** [comment]
If it's a use case that you think is really lacking in the doc feel free to do a PR about that :)

**Comment by Soullivaneuh** [comment]
> But why would you need to set an order?

Because some files contains instance that will be used by others. So I needed to load them first.

**Comment by theofidry** [comment]
Yes but it cant' work since the order is not reliable. That's why I change the `Loader` in the beta-4 to have multiple loading passes (default to 6 try). So if you have:

```
file1.yml -> requires file2
file2.yml -> requires file3
file3.yml
```

The first pass will load only file3, the second file 2 and the third file1.

**Comment by Soullivaneuh** [comment]
> in the beta-4 to have multiple loading passes

What do you mean? I can't follow you, can you rely the related documentation please?

**Comment by theofidry** [comment]
Let's say you have the following fixtures that will be loaded:

```
fixture_file1.yml
fixture_file2.yml
fixture_file3.yml
```

Where:

* `fixture_file1.yml` has references of fixtures declared in `fixture_file2.yml`
* `fixture_file2.yml` has references of fixtures declared in `fixture_file3.yml`

So *ideally*, we would have to load them in the following order:

```
1: fixture_file3.yml
2: fixture_file2.yml
3: fixture_file1.yml
```

But as I said, there is no reliable way for the order. So rather than trying to play with it, I changed the loader to try to load as many file as possible. Once done, the loaded objects are passed as references to the fixtures loader and it tries to load the files that could not be loaded. And it does so until it reaches the [loading limit](https://github.com/hautelook/AliceBundle/blob/e739300431db9fc9706006bde7b5e8d321c11a7d/src/Alice/DataFixtures/Loader.php#L45).

So in our example:

* first pass: load sucessfully `fixture_file3.yml` which returns a set of objects `$objects`
* second pass: try to reload `fixture_file1.yml` and `fixture_file2.yml` with `$objects` as references => successfully load  `fixture_file2.yml`. It returns a new set of objects merged to `$objects`.
* third pass: try to reload `fixture_file1.yml` with `$objects` as references => sucess

**Comment by theofidry** [comment]
No doc yet about that as it's a quick fix from #115.
